### PR TITLE
feat(dump): added project id to dump metadata url

### LINKED ISSUE #21: add project id to dump metadata url

### PR DESCRIPTION
**Pull request by melancholiai**
| Question                | Answer                                                                          |
| ---------------- | -------------------------------------------------------------------------- |
| Bug fix         | ✖                                                                        |
| New feature     | ✔                                                                        |
| Breaking change | ✖                                                                        |
| Deprecations    | ✖                                                                        |
| Documentation   | ✖                                                                        |
| Tests added     | ✔                                                                        |
| Chore            | ✖                                                                       |

Closes #21

### CODE REVIEW

**Review by syncush — CHANGES_REQUESTED** [review]
this needs to be covered with tests

**Comment by melancholiai** [comment]
> this needs to be covered with tests

it is, my bad for missing a V on the table

File: src/dumpMetadata/models/dumpMetadataManager.ts [hunk 1]
```diff
@@ -40,7 +40,10 @@ export class DumpMetadataManager {
   }
 
   private getUrlHeader(): string {
-    const { protocol, host, port } = this.objectStorageConfig;
+    const { protocol, host, projectId, port } = this.objectStorageConfig;
+    if (projectId != undefined) {
+      return `${protocol}://${host}/${projectId}:${port}`;
```
Comment (syncush) [inline src/dumpMetadata/models/dumpMetadataManager.ts:45]: what if `projectId` is empty, we would still have a semicolon, I do not think it is the desired behavior

**Review by syncush — CHANGES_REQUESTED** [review]

File: src/dumpMetadata/models/dumpMetadataManager.ts (hunk 1 above)
Comment (melancholiai) [reply src/dumpMetadata/models/dumpMetadataManager.ts:45]: I've added tests for the case `projectId` is not defined as well and changed the default value in `default.json` to be undefined.
in the case of an empty string the `projectId` will be accounted as undefined.

**Review by syncush — CHANGES_REQUESTED** [review]

File: src/dumpMetadata/models/dumpMetadataManager.ts [hunk 2]
```diff
@@ -40,7 +40,10 @@ export class DumpMetadataManager {
   }
 
   private getUrlHeader(): string {
-    const { protocol, host, port } = this.objectStorageConfig;
+    const { protocol, host, projectId, port } = this.objectStorageConfig;
+    if (projectId != undefined && projectId.length > 0) {
+      return `${protocol}://${host}/${projectId}:${port}`;
+    }
     return `${protocol}://${host}:${port}`;
   }
```
[the remark this answers is not in the document]
Comment (melancholiai) [reply src/dumpMetadata/models/dumpMetadataManager.ts]: done.

**Review by syncush — APPROVED** [review]

**Pull request merged**